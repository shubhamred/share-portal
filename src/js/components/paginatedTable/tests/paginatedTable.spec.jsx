import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Router } from 'react-router-dom';
import CustomPaginationActionsTable from '../paginatedTable';

configure({ adapter: new Adapter() });

// let CustomPaginationActionsTableWrapper;

describe('components/CustomPaginationActionsTable', () => {
  it(`match's snapshot`, () => {
    jest.mock('react-router-dom', () => ({
      useHistory: () => ({
        push: jest.fn()
      })
    }));
    const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
    const component = shallow(
      <Router history={historyMock}>
        <CustomPaginationActionsTable />
      </Router>
    );
    const tree = shallowToJson(component);
    expect(tree)
      .toMatchSnapshot();
    jest.clearAllMocks();
  });
  it(`Renders with Props`, () => {
    jest.mock('react-router-dom', () => ({
      useHistory: () => ({
        push: jest.fn()
      })
    }));
    const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
    const props = {
      dataTableType: 'deals',
      fetchNextData: jest.fn(),
      totalCount: 10,
      columns: ['Column 1', 'Column 2'],
      rowsPerPage: 10,
      setRowsPerPage: jest.fn(),
      page: 0,
      setPage: jest.fn(),
      rows: [{ column1: 'dummy 1' }, { column2: 'dummy 2' }],
      rowNames: ['column1', 'column2']
    };
    const component = shallow(
      <Router history={historyMock}>
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <CustomPaginationActionsTable {...props} />
      </Router>
    );
    const tree = shallowToJson(component);
    expect(tree)
      .toMatchSnapshot();
    jest.clearAllMocks();
  });
});
