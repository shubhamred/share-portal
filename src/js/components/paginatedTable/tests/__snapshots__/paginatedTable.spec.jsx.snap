// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CustomPaginationActionsTable Renders with Props 1`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "location": Object {},
        "push": [MockFunction],
      },
      "location": Object {},
      "match": Object {
        "isExact": false,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <ContextProvider
    value={
      Object {
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "location": Object {},
        "push": [MockFunction],
      }
    }
  >
    <CustomPaginationActionsTable
      checkedOptions={Array []}
      columns={
        Array [
          "Column 1",
          "Column 2",
        ]
      }
      dataTableType="deals"
      fetchNextData={[MockFunction]}
      handleCheckBoxValue={[Function]}
      handleClearDate={[Function]}
      handleClearStatus={[Function]}
      isPaginationRequired={true}
      onTableHeadClick={[Function]}
      page={0}
      rowNames={
        Array [
          "column1",
          "column2",
        ]
      }
      rows={
        Array [
          Object {
            "column1": "dummy 1",
          },
          Object {
            "column2": "dummy 2",
          },
        ]
      }
      rowsPerPage={10}
      selectedFromDate={Array []}
      setPage={[MockFunction]}
      setRowsPerPage={[MockFunction]}
      statusOptions={Array []}
      totalCount={10}
    />
  </ContextProvider>
</ContextProvider>
`;

exports[`components/CustomPaginationActionsTable match's snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "location": Object {},
        "push": [MockFunction],
      },
      "location": Object {},
      "match": Object {
        "isExact": false,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <ContextProvider
    value={
      Object {
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "location": Object {},
        "push": [MockFunction],
      }
    }
  >
    <CustomPaginationActionsTable
      checkedOptions={Array []}
      columns={Array []}
      dataTableType=""
      fetchNextData={[Function]}
      handleCheckBoxValue={[Function]}
      handleClearDate={[Function]}
      handleClearStatus={[Function]}
      isPaginationRequired={true}
      onTableHeadClick={[Function]}
      page={0}
      rowNames={Array []}
      rows={
        Array [
          Object {},
        ]
      }
      rowsPerPage={5}
      selectedFromDate={Array []}
      setPage={[Function]}
      setRowsPerPage={[Function]}
      statusOptions={Array []}
      totalCount={0}
    />
  </ContextProvider>
</ContextProvider>
`;
